flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/utils/PipelineTestOnYarnEnvironment.java

```diff
@@ -0,0 +1,327 @@
+            ex.printStackTrace();
+            LOG.error("setup failure", ex);
+            fail("");
+        }
+    }
+
+    // write yarn-site.xml to target/test-classes so that flink pick can pick up this when
+    // initializing YarnClient properly from classpath
+    public static void writeYarnSiteConfigXML(Configuration yarnConf, File targetFolder)
+            throws IOException {
+        yarnSiteXML = new File(targetFolder, "/yarn-site.xml");
+        try (FileWriter writer = new FileWriter(yarnSiteXML)) {
+            yarnConf.writeXml(writer);
+            writer.flush();
+        }
+    }
+
+    public String submitPipelineJob(String pipelineJob, Path... jars) throws Exception {
+        ProcessBuilder processBuilder = new ProcessBuilder();
+        Map<String, String> env = getEnv();
+        processBuilder.environment().putAll(env);
+        Path yamlScript = temporaryFolder.newFile("mysql-to-values.yml").toPath();
+        Files.write(yamlScript, pipelineJob.getBytes());
+
+        List<String> commandList = new ArrayList<>();
+        commandList.add(env.get("FLINK_CDC_HOME") + "/bin/flink-cdc.sh");
+        commandList.add("-t");
+        commandList.add("yarn-application");
+        commandList.add(yamlScript.toAbsolutePath().toString());
+        for (Path jar : jars) {
+            commandList.add("--jar");
+            commandList.add(jar.toString());
+        }
+
+        processBuilder.command(commandList);
+        LOG.info("starting flink-cdc task with flink on yarn-application");
+        Process process = processBuilder.start();
+        process.waitFor();
+        String applicationIdStr = getApplicationId(process);
+        Preconditions.checkNotNull(
+                applicationIdStr, "applicationId should not be null, please check logs");
+        ApplicationId applicationId = ApplicationId.fromString(applicationIdStr);
+        waitApplicationFinished(applicationId, yarnAppTerminateTimeout, sleepIntervalInMS);
+        LOG.info("started flink-cdc task with flink on yarn-application");
+        return applicationIdStr;
+    }
+
+    public Map<String, String> getEnv() {
+        Path flinkHome =
+                TestUtils.getResource(
+                        "flink-\\d+(\\.\\d+)*$",
+                        "flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/target");
+        Map<String, String> env = new HashMap<>();
+        env.put("FLINK_HOME", flinkHome.toString());
+        env.put("FLINK_CONF_DIR", flinkHome.resolve("conf").toString());
+        addFlinkConf(flinkHome.resolve("conf").resolve("config.yaml"));
+        Path flinkcdcHome =
+                TestUtils.getResource("flink-cdc-\\d+(\\.\\d+)*(-SNAPSHOT)?$", "flink-cdc-dist");
+        env.put("FLINK_CDC_HOME", flinkcdcHome.toString());
+        env.put("HADOOP_CLASSPATH", getYarnClasspath());
+        return env;
+    }
+
+    // TODO Maybe pipeline.yml should support adding flink conf
+    public void addFlinkConf(Path flinkConf) {
```

Comment:
We do this in another pull-request and remove TODO here.

Reply:
Done,Ditto.